commands/transaction.go

```diff
@@ -30,6 +33,34 @@ func NewInputData(height uint64) *commandspb.InputData {
 	}
 }
 
+func MarshalInputData(chainID string, inputData *commandspb.InputData) ([]byte, error) {
+	data, err := proto.Marshal(inputData)
+	if err != nil {
+		return nil, err
+	}
+
+	dataWithChainID := append([]byte(fmt.Sprintf("%s%c", chainID, ChainIDDelimiter)), data...)
+
+	return dataWithChainID, nil
+}
+
+func UnmarshalInputData(rawInputData []byte) (*commandspb.InputData, string, error) {
```

Comment:
Why not use bytes.HasPrefix, then slice the rest of it like so: return rawInputData[len(prefix):len(rawInputdata)]

Reply:
ok